forc-pkg/src/pkg.rs

```diff
@@ -2369,13 +2411,28 @@ pub fn build_with_options(build_options: BuildOpts) -> Result<Built> {
     };
 
     let manifest_file = ManifestFile::from_dir(&this_dir)?;
-    match manifest_file {
+    match &manifest_file {
         ManifestFile::Package(package_manifest) => {
-            let built_package = build_package_with_options(&package_manifest, build_options)?;
+            let built_package = build_package_with_options(package_manifest, build_options)?;
             Ok(Built::Package(Box::new(built_package)))
         }
-        ManifestFile::Workspace(_) => {
-            bail!("Workspace building is not supported")
+        ManifestFile::Workspace(workspace_manifest) => {
+            let lock_path = workspace_manifest.lock_path();
+            let member_manifests = manifest_file.member_manifests()?;
+            let build_plan = BuildPlan::from_lock_and_manifests(
+                &lock_path,
+                &member_manifests,
+                build_options.pkg.locked,
+                build_options.pkg.offline,
+            )?;
+            let manifest_map = build_plan.manifest_map();
+            let graph = build_plan.graph();
+            for member in build_plan.member_nodes() {
+                let pinned = &graph[member];
+                info!("Building {}", pinned.name);
+                build_package_with_options(&manifest_map[&pinned.id()], build_options.clone())?;
+            }
+            Ok(Built::Workspace)
```

Comment:
I think we should be returning all of the built packages as a part of the `Workspace` variant, perhaps within a `BuiltWorkspace` type.